Squeezer/src/main/java/uk/org/ngo/squeezer/Util.java

```diff
@@ -287,6 +296,39 @@ public static void crashlyticsLog(int priority, String tag, String msg) {
     public static void crashlyticsLogException(java.lang.Throwable throwable) {
         if (supportCrashlytics()) {
             Crashlytics.logException(throwable);
+        } else {
+            Log.i("Util.crashlyticsLog", "", throwable);
+        }
+    }
+
+    public static void moveFile(File sourceFile, File destinationFile) throws IOException {
+        File destFolder = destinationFile.getParentFile();
+        if (!destFolder.exists()) {
+            if (!destFolder.mkdirs()) {
+                throw new IOException("Cant create folder for '" + destinationFile + "'");
+            }
+        }
+        if (!sourceFile.renameTo(destinationFile)) {
+            // We could not rename. This may be because source and destination are on different
+            // mount points, so we attempt to copy and delete instead.
+            FileChannel sourceChannel = new FileInputStream(sourceFile).getChannel();
+            FileChannel destinationChannel = new FileOutputStream(destinationFile).getChannel();
+            try {
+                // Transfer the file in chunks to avoid out of memory issues
+                final long blockSize = Math.min(268435456, sourceChannel.size());
+                long position = 0;
+                while (destinationChannel.transferFrom(sourceChannel, position, blockSize) > 0) {
+                    position += blockSize;
+                }
+            } finally {
+                if (sourceChannel != null)
```

Comment:
According to the documentation getChannel does not return null, but the FileInputStream/FileOutputStream constructors may throw, in which case the corresponding channel is null.
The code was wrong however, as those constructors were called outside the try-finally block, so I fixed that.